core/src/test/scala/org/apache/spark/SparkConfSuite.scala

```diff
@@ -354,6 +354,20 @@ class SparkConfSuite extends SparkFunSuite with LocalSparkContext with ResetSyst
     }
   }
 
+  test("SPARK-26998: SSL configuration not needed on executors") {
+    val conf = new SparkConf(false)
+    conf.validateSettings()
+
+    conf.set("spark.ssl.enabled", "true")
+    conf.set("spark.ssl.keyPassword", "password")
+    conf.set("spark.ssl.keyStorePassword", "password")
+    conf.set("spark.ssl.trustStorePassword", "password")
+    conf.validateSettings()
+
+    val filtered = conf.getAll.filter { case (k, _) => SparkConf.isExecutorStartupConf(k) }
```

Comment:
To make the assert more fine-grained, you could `assert` that each isn't an executor conf with `foreach`

Reply:
The exception tells exactly what is the problematic parameter and value so not sure a more complex construct would help here:
```
Array((spark.network.foo,bar)) was not empty
ScalaTestFailureLocation: org.apache.spark.SparkConfSuite at (SparkConfSuite.scala:366)
org.scalatest.exceptions.TestFailedException: Array((spark.network.foo,bar)) was not empty
```

Reply:
OK that's fine